### ISSUE TITLE
Ability to respect VSCode indentation

### ISSUE THREAD

**Issue by probablykasper** [rust-analyzer/rust-analyzer#8974]
With VSCode's official Rust extension, rls formatting respects the configured VSCode indentation:

https://user-images.githubusercontent.com/11315492/119389040-5d9ec000-bccb-11eb-8023-089e6312be5f.mp4

Would really like to have that be possible with rust-analyzer

**Comment by bjorn3** [comment]
Rust-analyzer uses rustfmt for formatting. You can use `rustfmt.toml` to configure it. I think having rust-analyzer invoked rustfmt and `cargo fmt` result in different formatting is not a good idea.

**Comment by matklad** [comment]
Yeah, I agree with bjorn3 that rustfmt.toml should be the source of truth here. If anything, we should go the other way around and ask Code to indent with two spaces if that's specified in toml.

**Comment by probablykasper** [comment]
@bjorn3 I'm pretty sure the Rust extension uses rustfmt as well, right?

To be clear, I'm not suggesting this should override `rustfmt.toml`. Like the Rust extension, this would only apply when there is no `rustfmt.toml`

**Comment by lupuchard** [comment]
+1 for this
The [rust-analyzer extension's description](https://marketplace.visualstudio.com/items?itemName=rust-lang.rust-analyzer) doesn't mention that it affects formatting at all, let alone that it uses rustfmt. It's really annoying to have tabs configured both for the file and as the .vscode default and have that overwritten with no indication of what extension is doing it, when there isn't even a rustfmt.toml file.

At the very least rust-analyzer could create a rustfmt.toml file by default to indicate that this is overriding your indentation settings.

**Comment by ChayimFriedman2** [comment]
If you want to change indentation, you need to have rustfmt.toml. This is the standard across the Rust ecosystem.

r-a won't create rustfmt.toml, because this is already the default and r-a should not create files without user request.
